### PR TITLE
Added DP algorithms for Java Project

### PR DESCRIPTION
**Pull request by theDeepanshuMourya**
Added algorithms for Coin Change Problem  and Longest Increasing Subsequence

# Title/Heading

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generates no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: Java/CoinChange.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
+/*
+ * Dynamic Programming ALgorithm for Coin Change Problem.
+ * CoinChangeI:
+ * Given a value N, if we want to make change for N cents, and we have infinite supply of 
+ * each of C = { C1, C2, .. , Cm} valued coins, how many ways can we make the change? 
+ * The order of coins doesn’t matter.
+ * 
+ * Time complexity of this function: O(mn)
+ * Space Complexity of this function: O(n)
```
Comment (YashSaxena9) [inline Java/CoinChange.java]: Use capital letters to describe complexity. e.g.: O(N)

File: Java/CoinChange.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+            }
+
+            return (dp[amount] > amount) ? -1 : dp[amount];
+        }
+    }
+
+    public static void main(final String[] args) {
+        final int arr[] = { 1, 2, 3 };
+        final int n = arr.length;
+        final int m = 4;
```
Comment (YashSaxena9) [inline Java/CoinChange.java]: Please follow camelCase naming conventions for the code.

File: Java/LongestIncreasingSubsequence.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+ * long as possible. This subsequence is not necessarily contiguous, or unique.
+ */
+
+package Hacktoberfest2020_.Java;
+
+public class LongestIncreasingSubsequence {
+	public static class LIS_Memoization {
+		public static int lis(final int arr[]) {
+            final int n = arr.length;
+            final int lis[] = new int[n];
```
Comment (YashSaxena9) [inline Java/LongestIncreasingSubsequence.java]: Please follow camelCase naming conventions.

**Review by YashSaxena9 — CHANGES_REQUESTED** [review]
The code contributed to the repository matched all our requirements and we will be happy to merge it to our repository's main branch. Please refactor the variable names according to the camelCase naming conventions and also add a small docString(comment) on the top of the functions to describe their work.

File: Java/CoinChange.java (hunk 1 above)
Comment (theDeepanshuMourya) [reply Java/CoinChange.java]: Sure. I'll change that and submit the code again.

File: Java/CoinChange.java (hunk 2 above)
Comment (theDeepanshuMourya) [reply Java/CoinChange.java]: Sure. I'll change that and submit the code again.

File: Java/LongestIncreasingSubsequence.java (hunk 3 above)
Comment (theDeepanshuMourya) [reply Java/LongestIncreasingSubsequence.java]: Sure. I'll change that and submit the code again.

**Comment by theDeepanshuMourya** [comment]
Hi, I am beginner to all this. So can you just guide me that how do I merge the changes in this same pull request?

**Comment by YashSaxena9** [comment]
It is advisable to follow the camelCase naming convention for java codes and give a proper description of the use of the functions.

Follow the following article to know more about the best practices in naming conventions.
[Click Me!](https://medium.com/@rhamedy/a-short-summary-of-java-coding-best-practices-31283d0167d3)

**Comment by theDeepanshuMourya** [comment]
I have updated the code according to the changes you have suggested. Please check.

**Review by YashSaxena9 — APPROVED** [review]
Code reviewed

**Comment by YashSaxena9** [comment]
Merge Approved :)

**Pull request merged**